services/cloudapi/nsg/nsg.go

```diff
@@ -0,0 +1,93 @@
+package nsg
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	ionoscloud "github.com/ionos-cloud/sdk-go/v6"
+
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/cloudapi"
+)
+
+// Service implements utility methods for the Network Security Group
+type Service struct {
+	Client *ionoscloud.APIClient
+	Meta   interface{}
+	D      *schema.ResourceData
+}
+
+// PutServerNSG updates the security groups of a server
+func (nsg *Service) PutServerNSG(ctx context.Context, dcID, serverID string, rawIDs []any) diag.Diagnostics {
+	if dcID == "" || serverID == "" {
+		return diag.Errorf("dcID and serverID must be set")
+	}
+	if len(rawIDs) == 0 {
```

Comment:
@cristiGuranIonos what happens when I have a server with some `security_groups_ids` and then I want to empty the list and update the server, basically I want to remove the SG IDs for a server. From what I see, this change won't take effect since the function will return here. Let's double check this, maybe we have to remove this length check.

Reply:
removed, good catch